docs/src/user-docs/guides-using-object-storage/clp-usage.md

```diff
@@ -5,35 +5,90 @@ should be able to use CLP as described in the [clp-json quick-start guide](../qu
 
 ## Compressing logs from S3
 
-To compress logs from S3, use the `sbin/compress.sh` script as follows, replacing the fields in
+To compress logs from S3, use the `sbin/compress-from-s3.sh` script. The script supports two modes
+of operation. Select a mode by using the corresponding argument in the command:
+
+* [**s3-object** mode](#s3-object-compression-mode): Compress S3 objects specified by their full
+  S3 URLs.
+* [**s3-key-prefix** mode](#s3-key-prefix-compression-mode): Compress all S3 objects under a given
+  S3 key prefix.
+
+### `s3-object` compression mode
+
+`s3-object` mode allows you to specify individual S3 objects to compress by their full URLs. To
+use this mode, call the `sbin/compress-from-s3.sh` script as follows, and replace the fields in
 angle brackets (`<>`) with the appropriate values:
 
 ```bash
-sbin/compress.sh \
+sbin/compress-from-s3.sh \
   --timestamp-key <timestamp-key> \
-  <url>
+  --dataset <dataset-name> \
+  s3-object \
+  <object-url> [<object-url> ...]
 ```
 
-* `<url>` is a URL identifying the logs to compress. It can have one of two formats:
-  * `https://<bucket-name>.s3.<region-code>.amazonaws.com/<prefix>`
-  * `https://s3.<region-code>.amazonaws.com/<bucket-name>/<prefix>`
-* The fields in `<url>` are as follows:
+* `<object-url>` is a URL identifying the S3 object to compress. It can be written in either of two
+  formats:
+  * `https://<bucket-name>.s3.<region-code>.amazonaws.com/<object-key>`
+  * `https://s3.<region-code>.amazonaws.com/<bucket-name>/<object-key>`
+* The fields in `<object-url>` are as follows:
   * `<bucket-name>` is the name of the S3 bucket containing your logs.
   * `<region-code>` is the AWS region [code][aws-region-codes] for the S3 bucket containing your
     logs.
-  * `<prefix>` is the prefix of all logs you wish to compress and must begin with the
-    `<all-logs-prefix>` value from the [compression IAM policy][compression-iam-policy].
+  * `<object-key>` is the [object key][aws-s3-object-key] of the log file object you wish to
+    compress. There must be no duplicate object keys across all `<object-url>` arguments.
+
+Instead of typing the input object URLs explicitly in the command, you may specify them in a text
```

Comment:
```suggestion
* For a description of other fields, see the [clp-json quick-start
  guide](../quick-start/clp-json.md#compressing-json-logs).

Instead of specifying input object URLs explicitly in the command, you may specify them in a text
```